### PR TITLE
Add support for date comparison

### PR DESCRIPTION
**Pull request by tsop14**
Makes it possible to compare recipe publication date using quill by adding support for < and > operators. 
- Db context has been abstracted into a wrapper which defines the trait quote. 
- Context created then extends this trait. 
- Updates to UI and ETL to support change.

Thanks @LATaylor-guardian @tomrf1

### CODE REVIEW

File: etl/src/main/scala/etl/ETL.scala [hunk 1]
```diff
@@ -53,9 +54,12 @@ object ETL extends App {
     .showFields("main,body,byline")
     .showElements("image")
 
-  val dbContext = new JdbcContext[PostgresDialect, SnakeCase]("db.ctx")
+  val contextWrapper = new ContextWrapper { val config = ConfigFactory.load() }
+  val test = ConfigFactory.load()
```
Comment (tomrf1) [inline etl/src/main/scala/etl/ETL.scala]: debugging line?

File: common/src/main/scala/com/gu/recipeasy/db/ContextWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import io.getquill.{ ImplicitQuery, JdbcContext, PostgresDialect, SnakeCase }
+import java.time.OffsetDateTime
+
+trait ContextWrapper {
+  val config: TypesafeConfig
+  lazy val dbContext = new JdbcContext[PostgresDialect, SnakeCase](config.getConfig("db.ctx")) with ImplicitQuery with Quotes
+
+  trait Quotes {
+    this: JdbcContext[_, _] =>
+    implicit class DateTimeQuotes(left: OffsetDateTime) {
```
Comment (mchv) [inline common/src/main/scala/com/gu/recipeasy/db/ContextWrapper.scala]: need to extend `AnyVal`

**Comment by mchv** [comment]
Very nice workaround 💯

File: etl/src/main/scala/etl/ETL.scala (hunk 1 above)
Comment (tsop14) [inline etl/src/main/scala/etl/ETL.scala]: 👀 Quite right! Removing

File: common/src/main/scala/com/gu/recipeasy/db/ContextWrapper.scala (hunk 2 above)
Comment (tsop14) [inline common/src/main/scala/com/gu/recipeasy/db/ContextWrapper.scala]: Done

**Comment by tomrf1** [comment]
👍

**Comment by tsop14** [comment]
👍

**Pull request merged**